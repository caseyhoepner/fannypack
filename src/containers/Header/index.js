import React, { Component } from 'react';
import { connect } from 'react-redux';

import './Header.css'

export class Header extends Component {
  render() {
    return (
      <div className='h-header-container'>
        <h1 className='h-title'>FannyPack</h1>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
})

export default connect(null, mapDispatchToProps)(Header);