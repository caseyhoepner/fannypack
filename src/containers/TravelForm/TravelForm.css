.tf-form {
  width: 50%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  margin: auto;
}

.tf-title {
  font-size: 1.5rem;
  margin: 40px 0 20px;
}

.tf-input-component {
  height: 40px;
  border: none;
  margin-bottom: 5px;
  font-color: grey;
}

.tf-btn {
  margin-top: 20px;
  background-color: #E45C8F;
  border: none;
  width: 100%;
  border-radius: 25px;
  height: 50px;
  font-family: 'Shrikhand', cursive;
  font-size: 1.75rem;
}

.tf-form-components {
  width: 100%;
}

.tf-day-input {
  display: flex;
  align-items: center;
}

.tf-day {
  width: 100px;
}

.hide {
  display: none;
}

.tf-add-btn {
  color: #E45C8F;
}

.parent-container {
  height: 100vh;
}

.tf-form-components {
  font-color: grey;
}