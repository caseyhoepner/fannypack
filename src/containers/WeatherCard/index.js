import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { packingData } from '../../utils/packingData';
import './WeatherCard.css';

export const WeatherCard = ({ highTemp, lowTemp, summary, icon, city, day}) => {

  const getEssentials = () => {
    let essentials = [];

    if (highTemp >= 80) {
      essentials = [...essentials, ...packingData.hot]
    }

    if (highTemp < 80 && lowTemp > 70) {
      essentials = [...essentials, ...packingData.warm]
    }

    if (highTemp < 70 && lowTemp > 40) {
      essentials = [...essentials, ...packingData.cool]
    }

    if (lowTemp <= 40) {
      essentials = [...essentials, ...packingData.cold]
    }

    essentials = [...essentials, ...packingData[icon]];

    const filteredEssentials = filterEssentials(essentials)

    return filteredEssentials.map(filteredEssential => {
      return <li className='wc-essentials-list'key={filteredEssential}>{filteredEssential}</li>
      })
  }

  const filterEssentials = (essentials) => {
    return essentials.reduce((filteredEssentials, essential) => {
      if (!filteredEssentials.includes(essential)) {
        filteredEssentials.push(essential);
      }
      return filteredEssentials;
    }, []);
  }

  const formatCity = function (city) {
    city = city.toLowerCase().split(' ');
    for (var i = 0; i < city.length; i++) {
      city[i] = city[i].charAt(0).toUpperCase() + city[i].slice(1);
    }
    return city.join(' ');
  };

  return (
    <div className="wc-container">
      <div className='wc-weather-container'>
        <h2 className='wc-day'>{`Day ${day}: ${formatCity(city)}`}</h2>
          <section className='wc-temp-icon-container'>
            <img className='wc-icon' src={require(`../../assets/${icon}.svg`)} />
            <section className='wc-temp-container'>
              <p className='wc-high'>High: {highTemp.toFixed(0)}&#176;</p>
              <p className='wc-low'>Low: {lowTemp.toFixed(0)}&#176;</p>
            </section>
          </section>  
          <p className='wc-summary'>{summary}</p>
      </div>
      <section className='wc-essentials'>
        <h3 className='wc-essentials-title'>What to Pack:</h3>
          <ul className='wc-essentials-list'>
            { getEssentials() }
          </ul>
      </section>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  weatherData: state.weatherData
})

WeatherCard.propTypes = {
  weatherData: PropTypes.array
};

export default withRouter(connect(mapStateToProps)(WeatherCard));