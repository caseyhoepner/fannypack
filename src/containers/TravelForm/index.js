import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { setWeather, changeToLoaded } from '../../actions';
import Header from '../../containers/Header';
import * as API from '../../utils/';
import './TravelForm.css';

export class TravelForm extends Component {
  constructor() {
    super() 
    this.state = {
      day1: '',
      day2: '',
      day3: '',
      day4: '',
      day5: '',
      day6: '',
      day7: ''
    }
  }

  handleChange = (event) => {
    event.preventDefault();
    const { name, value } = event.target;
    this.setState({[name]: value});
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.history.push('/results');
    this.getWeatherData();
  }

  getWeatherData = async () => {
    const { day1, day2, day3, day4, day5, day6, day7 } = this.state;
    const { setWeather, changeToLoaded } = this.props;

    // for(let i = 1; i < 8; i++) {
    //   console.log(`day${i}`)
    //   if (`day${i}`) {
    //     const weatherData = await API.fetchWeather(`day${i}, ${i}`);
    //     setWeather(weatherData, `day${i}, ${i}`) 
    //   }
    // }

    if (day1) {
      const weatherData = await API.fetchWeather(day1, 1);
      setWeather(weatherData, day1, 1);
    }

    if (day2) {
      const weatherData = await API.fetchWeather(day2, 2);
      setWeather(weatherData, day2, 2);
    }

    if (day3) {
      const weatherData = await API.fetchWeather(day3, 3);
      setWeather(weatherData, day3, 3);
    }

    if (day4) {
      const weatherData = await API.fetchWeather(day4, 4);
      setWeather(weatherData, day4, 4);
    }

    if (day5) {
      const weatherData = await API.fetchWeather(day5, 5);
      setWeather(weatherData, day5, 5);
    }

    if (day6) {
      const weatherData = await API.fetchWeather(day6, 6);
      setWeather(weatherData, day6, 6);
    }

    if (day7) {
      const weatherData = await API.fetchWeather(day7, 7);
      setWeather(weatherData, day7, 7);
    }
    
    changeToLoaded(true);
  }

  render() {
    const { day1, day2, day3, day4, day5, day6, day7 } = this.state;

    return (
      <div className='parent-container'>
        <Header />
        <form className='tf-form'onSubmit={this.handleSubmit}>
          <h2 className='tf-title'>Where are you going?</h2>
            <div className='tf-form-components'>
              <section className='tf-tomorrow tf-day-input'>
                <p className='tf-day'>Tomorrow</p>
                  <input 
                    className='test-input tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day1'
                    value={day1}
                  />
              </section>
              <section 
                className='tf-day-2 tf-day-input'>
                <p className='tf-day'>Day 2</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day2'
                    value={day2}
                  />
              </section>              
              <section 
                className='tf-day-3 tf-day-input'>
                <p className='tf-day'>Day 3</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day3'
                    value={day3}
                  />
              </section>              
              <section className='tf-day-4 tf-day-input'>
                <p className='tf-day'>Day 4</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day4'
                    value={day4}
                  />
              </section>              
              <section className='tf-day-5 tf-day-input'>
                <p className='tf-day'>Day 5</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day5'
                    value={day5}
                  />
              </section>              
              <section className='tf-day-6 tf-day-input'>
                <p className='tf-day'>Day 6</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day6'
                    value={day6}
                  />
              </section>              
              <section className='tf-day-7 tf-day-input'>
                <p className='tf-day'>Day 7</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day7'
                    value={day7}
                  />
              </section>              
              <button 
                className='tf-form-component tf-btn'>
                Get Packing List
              </button>
            </div>
          </form>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setWeather: (weatherData, city, day) => dispatch(setWeather(weatherData, city, day)), 
  changeToLoaded: (bool) => dispatch(changeToLoaded(bool))
});

TravelForm.propTypes = {
  setWeather: PropTypes.func.isRequired,
  changeToLoaded: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(TravelForm);