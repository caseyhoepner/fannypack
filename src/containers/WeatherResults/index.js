import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Header from '../../components/Header';
import WeatherCard from '../../containers/WeatherCard';
import suitcase from '../../assets/suitcase.svg';
import './WeatherResults.css';
import  uuid from 'uuid';

export const WeatherResults = ({ weatherData, isLoaded }) => {
  let results;

  if (!isLoaded) {
    results = 
      <section className='wr-loading-container'>
        <img className='wr-suitcase' src={suitcase} alt='Icon of a suitcase.'/>
        <div className='typewriter'>
          <h1>Compiling your packing lists...</h1>
        </div>
      </section>
  } else {
    results = weatherData.map(city => {
      return <WeatherCard {...city} key={uuid()}/>;
    })
  };

    return (
      <div>
        <Header />
          <div className='wr-container'>
            { results }
          </div>
      </div>
    )
};

export const mapStateToProps = (state) => ({
  weatherData: state.weatherData,
  isLoaded: state.isLoaded
});

WeatherResults.propTypes = {
  weatherData: PropTypes.array.isRequired,
  isLoaded: PropTypes.bool.isRequired,
};

export default withRouter(connect(mapStateToProps)(WeatherResults));