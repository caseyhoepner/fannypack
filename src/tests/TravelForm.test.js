import React from 'react';
import { shallow } from 'enzyme';
import { TravelForm } from '../containers/TravelForm';

describe ('TravelForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TravelForm />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})