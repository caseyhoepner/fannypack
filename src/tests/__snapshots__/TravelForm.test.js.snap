// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TravelForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TravelForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="city"
          onChange={[Function]}
          placeholder="Destination"
          value=""
        />,
        <input
          name="date"
          onChange={[Function]}
          placeholder="Date of Travel"
          value=""
        />,
        <button>
          Get Packing List
        </button>,
      ],
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "city",
          "onChange": [Function],
          "placeholder": "Destination",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "date",
          "onChange": [Function],
          "placeholder": "Date of Travel",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get Packing List",
        },
        "ref": null,
        "rendered": "Get Packing List",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="city"
            onChange={[Function]}
            placeholder="Destination"
            value=""
          />,
          <input
            name="date"
            onChange={[Function]}
            placeholder="Date of Travel"
            value=""
          />,
          <button>
            Get Packing List
          </button>,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "city",
            "onChange": [Function],
            "placeholder": "Destination",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "date",
            "onChange": [Function],
            "placeholder": "Date of Travel",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get Packing List",
          },
          "ref": null,
          "rendered": "Get Packing List",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
